src/common/io.ts

```diff
@@ -1445,7 +1464,10 @@ export class ARIOWriteable extends ARIOReadable implements AoARIOWrite {
     params: AoIncreaseUndernameLimitParams,
     options?: WriteOptions,
   ): Promise<AoMessageResult> {
-    if (params.fundFrom === 'turbo') {
+    if (
+      this.paymentProvider instanceof TurboArNSPaymentProviderAuthenticated &&
```

Comment:
replace paymentProviders to be an Record<'turbo', PaymentProvider> and then modify this logic to do `this.paymentProvider[fundFrom].initiatiteArNSPurchase()`

Reply:
so this would translate to `this.paymentProviders['stakes' | 'balances' | 'any' | 'turbo'] ?

Reply:
you'd keep the `(if params.fundFrom == 'turbo')` but remove `this.paymentProvider instanceof TurboArNSPaymentProviderAuthenticated`

Reply:
this is a bit of a wierd state because we still need to ensure the payment provider is authenticated - due to the inconsistent signers, it might not be. Is the intention here to allow for easily selecting other payment providers in the future?